content/module2/6-traffic-routing.md

```diff
@@ -0,0 +1,112 @@
++++
+title = "Traffic routing"
+type = ""
+weight = 6
++++
+ 
+ # Reverse Proxy
+
+This is server that is used as the entrypoint to your machine, using then the request **Host** and **Path** requested (layer 7) to route traffic to the appropriate machine. 
+Let's imagine you have multiple servers running in your LAN. You would expose your reverse proxy to the internet, that proxy would receive all request and the route then accordingly. For example `kalschatzi.com` routes to one internal IP and port, and `learn.kalschtzi.com` to another different pair of IP and port.
+
+
+```mermaid
+flowchart TB
+
+subgraph Reverse_Proxy[Reverse Proxy]
+    C2[Client] --> RP[Reverse Proxy]
+    RP --> S1[Backend Server 1]
+    RP --> S2[Backend Server 2]
+end
+
+C2:::client
+RP:::proxy
+
+classDef client fill:#cce5ff,stroke:#000,stroke-width:1px
+classDef proxy fill:#ffeeba,stroke:#000,stroke-width:1px
+
+```
+
+## Forward proxy
+This does the inverse where it hides the **client** rather than the server. The way that it works is that when entering a URL on a browser like `kalschatzi.com` it first connects to the configured proxy server and only then to the actual `kalschatzi.com` server, which can itself be exposed with a reverse proxy. This is a **client** side configuration and will not affect in any way the server
+
+### Why use one?
+There can be a number of benefits, depending on need.
+* **Caching**: proxies can cache GET requests and this speeds up responses and saves bandwidth
+* **Content Filtering / Control**: It allows content being request to be filtered and blocked if necessary.
+* **Privacy**: It can hide your real IP address.
+* **Security**: It can enforce security policies like only allow HTTPS sites
+
+```mermaid
+flowchart TB
+
+subgraph Forward_Proxy[Forward Proxy]
+    C1[Client] --> FP[Forward Proxy]
+    FP --> Internet[Public Internet]
+    Internet --> Site1[example.com]
+end
+
+C1:::client
+FP:::proxy
+
+classDef client fill:#cce5ff,stroke:#000,stroke-width:1px
+classDef proxy fill:#ffeeba,stroke:#000,stroke-width:1px
+
+```
+
+## Load Balancers
+
+Load balancers are similar to reverse proxies with a bigger focus on traffic shapping. Their main goal is ensure high availability and load distribution.
```

Comment:
Pedantic comment :) . I would put the acronym here - "Load Balancers(LBs)" since you are using the acronym later on.